nisystemlink/clients/artifact/_artifact_client.py

```diff
@@ -81,3 +86,13 @@ def download_artifact(self, id: Path) -> IteratorFileLike:
 
         """
         ...
+
+    @delete("artifacts/{id}")
+    def delete_artifact(self, id: Path) -> None:

```

Comment:
We prefer to keep the uplink types out of the function parameter type hints.
```suggestion
    @delete("artifacts/{id}", args[Path])
    def delete_artifact(self, id: str) -> None:
```